adam_core/coordinates/cometary.py

```diff
@@ -491,43 +298,77 @@ def from_cartesian(
+
+    def to_keplerian(self) -> "KeplerianCoordinates":
+        from .keplerian import KeplerianCoordinates
+
+        return KeplerianCoordinates.from_cartesian(self.to_cartesian())
+
+    @classmethod
+    def from_keplerian(
+        cls, keplerian_coordinates: "KeplerianCoordinates"
+    ) -> "CometaryCoordinates":
+        return cls.from_cartesian(keplerian_coordinates.to_cartesian())
+
+    def to_spherical(self) -> "SphericalCoordinates":
+        from .spherical import SphericalCoordinates
+
+        return SphericalCoordinates.from_cartesian(self.to_cartesian())
+
     @classmethod
-    def from_df(
-        cls: Type["CometaryCoordinates"],
-        df: pd.DataFrame,
-        coord_cols: dict = COMETARY_COLS,
-        origin_col: str = "origin",
-        frame_col: str = "frame",
+    def from_spherical(
+        cls, spherical_coordinates: "SphericalCoordinates"
     ) -> "CometaryCoordinates":
+        return cls.from_cartesian(spherical_coordinates.to_cartesian())
+
+    def to_dataframe(
+        self, sigmas: bool = False, covariances: bool = True
+    ) -> pd.DataFrame:
         """
-        Create a KeplerianCoordinates class from a dataframe.
+        Convert coordinates to a pandas DataFrame.
 
         Parameters
         ----------
-        df : `~pandas.DataFrame`
-            Pandas DataFrame containing Keplerian coordinates and optionally their
-            times and covariances.
-        coord_cols : dict
-            Dictionary containing as keys the coordinate dimensions and their equivalent columns
-            as values. For example,
-                coord_cols = {}
-                coord_cols["q"] = Column name of periapsis distance values
-                coord_cols["e"] = Column name of eccentricity values
-                coord_cols["i"] = Column name of inclination values
-                coord_cols["raan"] = Column name of longitude of ascending node values
-                coord_cols["ap"] = Column name of argument of periapsis values
-                coord_cols["tp"] = Column name of time of periapsis passage values.
-        origin_col : str
-            Name of the column containing the origin of each coordinate.
-        frame_col : str
-            Name of the column containing the coordinate frame.
-        """
-        data = Coordinates._dict_from_df(
-            df, coord_cols=coord_cols, origin_col=origin_col, frame_col=frame_col
+        sigmas : bool, optional
+            If True, include 1-sigma uncertainties in the DataFrame.
+        covariances : bool, optional
+            If True, include covariance matrices in the DataFrame. Covariance matrices
+            will be split into 21 columns, with the lower triangular elements stored.
+
+        Returns
+        -------
+        df : `~pandas.Dataframe`
+            DataFrame containing coordinates.
+        """
+        return coords_to_dataframe(
+            self, list(COMETARY_COLS.keys()), sigmas=sigmas, covariances=covariances
```

Comment:
Same comment here as for CartesianCoordinates' dataframes methods and the COLS thing.